packages/markdown/components/Embed/index.jsx

```diff
@@ -0,0 +1,78 @@
+/* eslint-disable react/jsx-props-no-spreading, jsx-a11y/iframe-has-title */
+const React = require('react');
+const propTypes = require('prop-types');
+
+const Favicon = ({ src, alt = 'favicon', ...attr }) => <img {...attr} alt={alt} height="14" src={src} width="14" />;
+Favicon.propTypes = {
+  alt: propTypes.string,
+  src: propTypes.string,
+};
+
+class Embed extends React.Component {
+  render() {
+    const { provider, url, title, html, iframe, image, favicon, ...attrs } = this.props;
+
+    if ('iframe' in this.props) {
+      return <iframe {...attrs} border="none" src={url} style={{ border: 'none' }} />;
+    }
+
+    const classes = ['embed', image && 'embed_hasImg'];
+    return (
+      <div className={classes.join(' ')}>
+        {html ? (
+          <div className="embed-media" dangerouslySetInnerHTML={{ __html: html }}></div>
+        ) : (
+          <a className="embed-link" href={url} rel="noopener noreferrer" target="_blank">
+            {!image || <img alt={title} className="embed-img" loading="lazy" src={image} />}
+            {title ? (
+              <div className="embed-body">
+                {!favicon || (
+                  <Favicon
+                    alt={provider}
+                    className="embed-favicon"
+                    loading="lazy"
+                    src={favicon}
+                    style={{ float: 'left' }}
+                  />
+                )}
+                {provider && (
+                  <small className="embed-provider">
+                    {provider.search(/^@{1}/) < 0 ? (
+                      provider
+                    ) : (
+                      <code style={{ fontFamily: 'var(--md-code-font, monospace)' }}>{url}</code>
```

Comment:
Does the `var` here work automatically, or does this need to be in a CSS file?

Reply:
Style attributes are pure CSS, so variable properties can be applied _and_ defined on elements inline wherevs!

Reply:
neat

Reply:
And wrt/best practices, yeah—generally we should prefer stylesheets to style attributes. But for single declarations and one-off markup I'm okay with breaking this rule!